hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/result/undelivered/LogUndeliveredMessageHandlerSpec.groovy

```diff
@@ -0,0 +1,58 @@
+package pl.allegro.tech.hermes.consumers.consumer.result.undelivered
+
+import pl.allegro.tech.hermes.api.Subscription
+import pl.allegro.tech.hermes.common.config.ConfigFactory
+import pl.allegro.tech.hermes.common.config.Configs
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException
+import pl.allegro.tech.hermes.common.kafka.KafkaTopicName
+import pl.allegro.tech.hermes.common.kafka.offset.PartitionOffset
+import pl.allegro.tech.hermes.common.message.undelivered.UndeliveredMessageLog
+import pl.allegro.tech.hermes.consumers.consumer.Message
+import pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+import java.time.Clock
+
+import static pl.allegro.tech.hermes.api.SentMessageTrace.createUndeliveredMessage
+import static pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult.failedResult
+import static pl.allegro.tech.hermes.consumers.test.MessageBuilder.withTestMessage
+
+
+class LogUndeliveredMessageHandlerSpec extends Specification {
+
+    private static final String MESSAGE_CONTENT = "ala ma kota"
+    private static final long TIME = 732648
+    private static final String CLUSTER = "test_cluster"
+    private static final long OFFSET = 124412L;
+    private static final int PARTITION = 5;
+
+    ConfigFactory configFactory = Stub()
+    Clock clock = Stub()
+    UndeliveredMessageLog undeliveredMessageLog = Mock()
+
+    Message message = withTestMessage()
+            .withContent(MESSAGE_CONTENT, StandardCharsets.UTF_8)
+            .withPartitionOffset(new PartitionOffset(KafkaTopicName.valueOf("kafka_topic"),
+                                            OFFSET, PARTITION))
+            .build()
+
+    Subscription subscription = Stub()
+
+    InternalProcessingException cause = new InternalProcessingException("Test cause.");
+    MessageSendingResult messageSendingResult = failedResult(cause);
+
+    def "should log message to undeliveredMessageLog"() {
```

Comment:
okay, so this test has little merit, it does more stubbing and mocking just to see if a mocked line is called - i think we have integration tests that guard logging and they cover it better